src/test/bind/test_bind.c

```diff
@@ -58,175 +26,94 @@ static int _do_bind(int fd, in_addr_t address, in_port_t port) {
     bindaddr.sin_port = port;
 
     /* bind the socket to the server port */
-    int result = bind(fd, (struct sockaddr *) &bindaddr, sizeof(struct sockaddr_in));
-
-    MYLOG("bind() returned %i", result);
-
-    if (result < 0) {
-        MYLOG("bind() error was: %s", strerror(errno));
-        return EXIT_FAILURE;
-    }
-
-    return EXIT_SUCCESS;
-}
-
-static int _do_listen(int fd) {
-    int result = listen(fd, 0);
-
-    MYLOG("listen() returned %i", result);
-
-    if (result < 0) {
-     MYLOG("listen() error was: %s", strerror(errno));
-     return EXIT_FAILURE;
-    }
-
-    return EXIT_SUCCESS;
+    return bind(fd, (struct sockaddr *) &bindaddr, sizeof(struct sockaddr_in));
 }
 
 static int _do_connect(int fd, struct sockaddr_in* serveraddr) {
     int count = 0;
+    int saved_errno = 0;
+    int result = 0;
 
-    while(1) {
-        int result = connect(fd, (struct sockaddr *) serveraddr, sizeof(struct sockaddr_in));
-
-        MYLOG("connect() returned %i", result);
-
-        if (result < 0 && errno == EINPROGRESS) {
-            MYLOG("connect() returned EINPROGRESS, retrying in 1 millisecond");
-            usleep((useconds_t)1000);
-
-            count++;
-
-            if(count > 1000) {
-                MYLOG("waited for accept for 1 second, giving up");
-                return EXIT_FAILURE;
-            }
-
-            continue;
-        } else if (result < 0) {
-            MYLOG("connect() error was: %s", strerror(errno));
-            return EXIT_FAILURE;
-        } else {
+    while (1) {
+        result = connect(fd, (struct sockaddr*)serveraddr,
+                         sizeof(struct sockaddr_in));
+        saved_errno = errno;
+        if (result == 0 || errno != EINPROGRESS) {
+            break;
+        }
+        if (count++ > 1000) {
+            g_message("waited for connect for 1 second, giving up");
             break;
         }
+        g_debug("connect() returned EINPROGRESS, retrying in 1 millisecond");
+        usleep((useconds_t)1000);
     }
-
-    return EXIT_SUCCESS;
+    errno = saved_errno;
+    return result;
 }
 
-static int _do_accept(int fd, int* outfd) {
+static int _do_accept(int fd) {
```

Comment:
I wonder if any of the bind/connect/accept helpers can or should be extracted to the `src/test/test_common.h` file?

Reply:
I looked a bit at doing this, but I'm not sure they make sense in a wider context. Leaving it for now.